### ISSUE TITLE
Feature request: Support of workingLocations / eventTypes

### ISSUE THREAD

**Issue by IT-Fuzzi** [allenporter/gcal_sync#426]
I'm searching for a way to query a Google Calendar for the eventType **workingLocation**, which is meanwhile supported by the official GoogleCalendar API 3.  [Google Documentation - Calendar Status](https://developers.google.com/calendar/api/guides/calendar-status)

![image](https://github.com/allenporter/gcal_sync/assets/34891422/865eb129-142f-4b7b-a65b-7a0b11f1be75)

A workingLocation can be specified as:

- homeOffice
- officeLocation
- customLocation (with own label)

I would like to use that within Home Assitant automations, e.g. to turn on the floor heating if I'm the home office.

**Comment by allenporter** [comment]
Agreed this would be a useful feature to have and I saw the email when they announced adding this. My impression is that is not available for consumer accounts (but is available for enterprise accounts) which is why I have not tried to add this yet.

**Comment by IT-Fuzzi** [comment]
Yes, it‘s only available for workspace/enterprise google accounts, not for private ones. 

I‘m not 100 % sure but I assume the Google API is supporting it since a few months. However, I also re-read the announcement mail which was sent a three days ago around the event types / the change to the Calendar API. Within this Google mentioned:

>Events.list: Specify the event types you need for your application as a parameter. Without making this change, working location events will start to be returned. 



So I assume that as of March 11, the workingLocation might even be returned if you do not explicitly query for it. Which might be a downside for certain people, while some (like me) may even benefit without any change in the integration. Might be that I was even too quick with my feature request, but sure a filter option would be appreciated nevertheless.

**Comment by allenporter** [comment]
I don't really think it's necessary to add home assistant configuration options for this since it's possible to make automations with calendar triggers match or exclude different events type.  Passing through additional work location events seems fine to me.

We can add the event type option in the library and home assistant could start requesting these.  I just won't be able to verify myself.

I'm not sure what happens with existing synced calendar events and if these will also be synced or if the resume token needs to be thrown away since the request parameters are changing.
